src/Agent.Listener/Environment.cs

```diff
@@ -0,0 +1,338 @@
+                { "curl" },
+                { "git" },
+                { "gulp" },
+                { "java" },
+                { "JDK", "javac" },
+                { "make" },
+                { "maven", "mvn" },
+                { "MSBuild", "xbuild" },
+                { "node.js", "node" },
+                { "node.js", "nodejs" },
+                { "npm" },
+                { "python" },
+                { "python3" },
+                { "sh" },
+                { "subversion", "svn" },
+                { "ruby" },
+                { "rake" },
+                { "Xamarin.iOS", "mdtool", new string[] { "/Applications/Xamarin Studio.app/Contents/MacOS/mdtool" } },
+                { "Xamarin.Android", "mandroid", new string[] { "/Library/Frameworks/Xamarin.Android.framework/Commands/mandroid" } }
+            };
+
+        private static readonly List<ShellCapability> _shellCapabilities =
+            new List<ShellCapability>
+            {
+                { "xcode", "xcode-select", "-p" },
+            };
+
+        private static readonly string[] _ignoredEnvVariables = new string[] {
+            "TERM_PROGRAM",
+            "TERM",
+            "TERM_PROGRAM_VERSION",
+            "SHLVL",
+            "ls_colors",
+            "comp_wordbreaks"
+        };
+
+        // Ignore env vars specified in the 'VSO_AGENT_IGNORE' env var
+        private const string EnvIgnore = "VSO_AGENT_IGNORE";
+
+        private Dictionary<string, string> _capsCache;
+
+        public async Task<Dictionary<string, string>> GetCapabilities(string agentName, CancellationToken token)
+        {
+            if (_capsCache != null)
+            {
+                return new Dictionary<string, string>(_capsCache);
+            }
+
+            var caps = new Dictionary<string, string>();
+
+            GetFilteredEnvironmentVars(caps);
+
+            GetRegularCapabilities(caps, token);
+
+            await GetShellCapabilities(caps, token);
+
+            caps["Agent.Name"] = agentName ?? string.Empty;
+            //TODO: figure out what should be the value of Agent.OS
+            //XPLAT is using process.platform, which returns 'darwin', 'freebsd', 'linux', 'sunos' or 'win32'
+            //windows agent is printing environment variable "OS", which is something like "Windows_NT" even when running Windows 10
+            //.Net core API RuntimeInformation.OSDescription is returning "Microsoft Windows 10.0.10586", 
+            //"Linux 3.13.0-43-generic #72-Ubuntu SMP Mon Dec 8 19:35:06 UTC 2014", "Darwin 15.4.0 Darwin Kernel Version 15.4.0: Fri Feb 26 22:08:05 PST 2016;"
+            caps["Agent.OS"] = RuntimeInformation.OSDescription ?? string.Empty;
+            caps["Agent.ComputerName"] = System.Environment.MachineName ?? string.Empty;
+
+            foreach (var cap in caps)
+            {
+                Trace.Info($"Capability: {cap.Key} Value: {cap.Value}");
+            }
+
+            _capsCache = new Dictionary<string, string>(caps);
```

Comment:
Should OrdinalIgnoreCase be used? Is this required if we merge the .env file? Should this be platform specific? I think on the server it's case insensitive.